pkg/rbd/rbd.go

```diff
@@ -0,0 +1,57 @@
+package rbd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// List retrieves and displays the Ceph block pools and their associated images.
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) {
+	blockPoolList, err := clientsets.Rook.CephV1().CephBlockPools(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(err)
+	}
+	// var blockPoolNames []string
+	var blockPoolNames map[string]string
+	blockPoolNames = make(map[string]string)
+
+	for _, blockPool := range blockPoolList.Items {
+		// blockPoolNames = append(blockPoolNames, blockPool.Name)
+		blockPoolNames[blockPool.Name] = "--"
+	}
+
+	cmd := "rbd"
+	for poolName, _ := range blockPoolNames {
+		args := []string{"ls", "--pool=" + poolName}
+		list, err := exec.RunCommandInOperatorPod(ctx, clientsets, cmd, args, clusterNamespace, clusterNamespace, true)
+		if err == nil && len(list) > 0 {
+			blockPoolNames[poolName] = strings.ReplaceAll(list, "\n", "")
+		}
+	}
+	PrintBlockPoolNames(blockPoolNames)
+}
+
+// PrintBlockPoolNames takes a map of block pool names to image names and prints them in a tabular format.
+func PrintBlockPoolNames(blockPoolNames map[string]string) {
+	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
+
+	// Print header with column names
+	fmt.Fprintln(writer, "poolName\timageName\t")
+	fmt.Fprintln(writer, "--------\t---------\t")
+
+	// Print each row from the map
+	for poolName, imageName := range blockPoolNames {
+		fmt.Fprintf(writer, "%s\t%s\t\n", poolName, imageName)
+	}
+
+	// Flush the writer
+	writer.Flush()
```

Comment:
```suggestion
	defer writer.Flush()
```
 and put this where writer is initialized at L45

Reply:
done